core/states/search_pattern.py

```diff
@@ -36,6 +36,9 @@ def on_event(self, event) -> RoverState:
         elif event == core.AutonomyEvents.ABORT:
             state = core.states.Idle()
 
+        elif event == core.AutonomyEvents.SEARCH_RADIUS_EXCEEDED:
```

Comment:
You will need to update docs/state_machine_matrix.csv to include this new event. Otherwise the unit tests will keep breaking.